src/lib/charts/timeSeriesDisplayToChartConfig.ts

```diff
@@ -16,14 +17,33 @@ import {
   AxisPosition,
   AxisType,
 } from '@deltares/fews-web-oc-charts'
+import { convertFewsPiDateTimeToJsDate } from '../date/index.js'
 
 export function timeSeriesDisplayToChartConfig(
   subplot: TimeSeriesDisplaySubplot,
   title: string,
+  period?: ActionPeriod,
 ): ChartConfig {
+  const xAxis = subplot.xAxis ? xAxisFromPlotItemXAxis(subplot.xAxis) : []
+  if (period) {
+    // The period is always specified in UTC.
+    const timeZoneOffsetString = 'Z'
+    const domain: [Date, Date] = [
+      convertFewsPiDateTimeToJsDate(period.startDate, timeZoneOffsetString),
+      convertFewsPiDateTimeToJsDate(period.endDate, timeZoneOffsetString),
+    ]
+    // Set the domain of the x-axis to be equal to the period, even if there are
+    // no data points in parts of the period.
+    if (xAxis.length === 0) {
+      xAxis.push({ domain })
+    } else {
+      xAxis[0].domain = domain
```

Comment:
Is only the first xAxis relevant?

Reply:
Yes, for time series there will always be just the one x-axis. I'll add a comment.